snakemake/executors.py

```diff
@@ -2052,3 +2059,605 @@ def run_wrapper(
+            )
+            lookup[zone["name"]] = self._retry_request(request)["items"]
+
+        # Only keep those that are shared, use last zone as a base
+        machineTypes = {mt["name"]: mt for mt in lookup[zone["name"]]}
+        del lookup[zone["name"]]
+
+        # Update final list based on the remaining
+        to_remove = set()
+        for zone, types in lookup.items():
+            names = [x["name"] for x in types]
+            for machineType in list(machineTypes.keys()):
+                if machineType not in names:
+                    to_remove.add(machineType)
+
+        for machineType in to_remove:
+            del machineTypes[machineType]
+
+        # Alert the user of machineTypes available before filtering
+        # https://cloud.google.com/compute/docs/machine-types
+        logger.debug(
+            "found {} machine types across regions {} before filtering"
+            "to increase selection, define fewer regions".format(
+                len(machineTypes), self.regions
+            )
+        )
+
+        # First pass - eliminate anything that too low in cpu/memory
+        keepers = dict()
+        for name, machineType in machineTypes.items():
+            if machineType["guestCpus"] < cores or machineType["memoryMb"] < mem_mb:
+                continue
+            keepers[name] = machineType
+
+        # If a prefix is set, filter down to it
+        if self._machine_type_prefix:
+            machineTypes = keepers
+            keepers = dict()
+            for name, machineType in machineTypes.items():
+                if name.startswith(self._machine_type_prefix):
+                    keepers[name] = machineType
+
+        # If we don't have any contenders, workflow error
+        if not keepers:
+            raise WorkflowError(
+                "Machine prefix {prefix} is too strict, and there are no options "
+                "available.".format(prefix=self._machine_type_prefix)
+            )
+
+        # Now find (quasi) minimal to satisfy constraints
+        machineTypes = keepers
+        keepers = dict()
+
+        minCores = 1
+        minMem = 3.75
+        smallest = "n1-standard-1"
+
+        for name, machineType in machineTypes.items():
+            if machineType["guestCpus"] < minCores and machineType["memoryMb"] < minMem:
+                smallest = name
+                minCores = machineType["guestCpus"]
+                minMem = machineType["memoryMb"]
+
+        selected = machineTypes[smallest]
+        logger.debug(
+            "Selected machine type {}:{}".format(smallest, selected["description"])
+        )
+        virtualMachine = {
+            "machineType": smallest,
+            "labels": {"app": "snakemake"},
+            "bootDiskSizeGb": 100,  # default is likely 10
+        }
+
+        resources = {"regions": self.regions, "virtualMachine": virtualMachine}
+        return resources
+
+    def _generate_build_package(self):
```

Comment:
Package is a very generic name. Maybe `source_package`?

Reply:
Sure thing.